api/rds.proto

```diff
@@ -201,8 +201,24 @@ message RouteAction {
       // random.
       string header_name = 1;
     }
+    message Cookie {
+      // The name of the cookie that will be used to obtain the hash key. If the
+      // cookie is not present and ttl below is not set, the load balancer will
+      // use a random number as the hash, effectively making the load balancing
+      // policy random.
+      string name = 1;
+      // If specified, a cookie will be generated if not presented with the
+      // specified TTL.
+      google.protobuf.Duration ttl = 2;
```

Comment:
So there are two kinds of cookie hashing:

1. You take a cookie that's present in the cookies header, you hash on its value.

2. Generated cookies. The proxy generates and sets a cookie with an expiration (TTL) on the first request from the client in its response to the client, based on the endpoint the request gets sent to. The client then presents this on the next and all subsequent requests, the hash of this is sufficient to ensure these requests get sent to the same endpoint.

This is a feature in Google's external L7 load balancer, https://cloud.google.com/compute/docs/load-balancing/http/backend-service#generated_cookie_affinity, as a concrete example of how it is used.

FWIW, this is useless with gRPC, since gRPC clients don't implement cookie stores. It's mostly useful for web traffic backed by a real browser.